docs/modules/ROOT/pages/high_level/security_group_host_design.adoc

```diff
@@ -0,0 +1,466 @@
+and requires the IP Tables chain to be recompiled (TBD: review Neutron Agent behavior)
+4. This IP Tables implementation cannot be offloaded to hardware, except using a SoC/ARM based NIC which has capability to run the whole Linux OS
+
+[source,shell]
+------------------------------------------------------------
+   +-------------+                     +---------------+
+   |             |                     |               |
+   |    VM 1     |                     |     VM 2      |
+   |             |                     |               |
+   +---+------+--+                     +---+-------+---+
+       | eth  |                            |  eth  |
+       +--+---+                            +---+---+
+          |                                    |
+       +--+---+                            +---+---+
+       | tap  |                            |  tap  |
+ +-----+------+-----+                 +----+-------+-----+
+ |                  |                 |                  |
+ |   Linux Bridge 1 |                 |   Linux Bridge 2 |
+ |                  |                 |                  |
+ +-----+------+-----+                 +-----+------+-----+
+       | veth |                             | veth |
+       +--+---+                             +--+---+
+          |                                    |
+       +--+---+                             +--+---+
+       | veth |                             | veth |
++------+------+-----------------------------+------+---------+
+|                                                            |
+|                     OVS  Bridge  (br-int)                  |
+|                                                            |
++------------------------------------------------------------+
+------------------------------------------------------------
+
+
+=== After OpenStack Mitaka Release
+
+The OpenStack community recognized the challenge of IP Tables implementation for security group rules.
+With OVS implementation of connection tracking <<ovs-conntrack>>, many open source solutions including ovs_dpdk_firewall<<intel>> and Dragonflow<<dragonflow-security-group>> start leveraging it for OpenStack.
+It is natural for OpenStack Neutron to adopt it.
+OpenStack Mitaka introduced official OVS Firewall driver, which addressed many of the issues in the prior releases.
+But Alcor can do even better than that.
+
+
+== Design approach
+
+The Security Group design uses OVS and OpenFlow, building on the latest Neutron OVS firewall rule approach.
+It offers the following advantages compared to Neutron:
+
+. On-Host direct connection from VM port to OVS: it doesn't need extra Linux bridge, Linux devices, Linux kernel or IP Tables;
+. Direct and simplified communication between Alcor controller to Alcor control agent;
+. Delta update: Update of security group rule only require the changed rule to be push down to compute hosts;
```

Comment:
yes, we will need delta update for ovs flows also. In order to support that, security group implementation in ACA will need to keep an internal map of port to security group (and its rules) mapping. Keep it in runtime memory would work, but it will not survive reboot/crash and hard to support older ACA to new ACA update (note that this problem is also applicable to ther ACA modules which is runtime memory as storage).

Reply:
Let's step back to the advantage of ACA vs neutron OVS agent:
1. plan to survive ACA crash or migration from older ACA to new ACA
2. delta update when only one security group rule is added/deleted/modified

Both of them requires ACA to keep an internal map of port to security group (and its rules) mapping. With this internal map, we can accept delta update and figure out the minimal set of openflow rule to modify.